File: clients/rust/src/instructions/sync_integration/sync_integration.rs
use crate::{derive_controller_authority_pda, generated::instructions::SyncBuilder};
use solana_instruction::Instruction;
use solana_pubkey::Pubkey;

pub fn create_sync_integration_instruction(
    controller: &Pubkey,
    payer: &Pubkey,
    integration: &Pubkey,
    reserve: &Pubkey,
) -> Instruction {
    let controller_authority = derive_controller_authority_pda(controller);

    SyncBuilder::new()
        .controller(*controller)
        .controller_authority(controller_authority)
        .payer(*payer)
        .integration(*integration)
        .reserve(*reserve)
        .instruction()
}

File: clients/ts/src/instructions/syncIntegration.ts
import { Address, TransactionSigner } from "@solana/kit";
import { getSyncInstruction } from "../generated/instructions/sync";
import { deriveControllerAuthorityPda } from "../pdas";
import { SVM_ALM_CONTROLLER_PROGRAM_ADDRESS } from "../generated";

/**
 * Instruction generation for syncing an integration account
 */
export async function createSyncIntegrationInstruction(
  controller: Address,
  payer: TransactionSigner,
  integration: Address,
  reserve: Address
) {
  const controllerAuthority = await deriveControllerAuthorityPda(controller);

  return getSyncInstruction({
    controller,
    controllerAuthority,
    payer,
    integration,
    reserve,
  });
}

